modules/bullet/area_bullet.h

```diff
@@ -0,0 +1,171 @@
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef AREABULLET_H
+#define AREABULLET_H
+
+#include "collision_object_bullet.h"
+#include "core/vector.h"
+#include "servers/physics_server.h"
+
+class SpaceBullet;
+class btGhostObject;
+
+class AreaBullet : public CollisionObjectBullet {
+	friend void SpaceBullet::check_ghost_overlaps();
+
+public:
+	struct InOutEventCallback {
+		ObjectID event_callback_id;
+		StringName event_callback_method;
+
+		InOutEventCallback()
+			: event_callback_id(0) {}
+	};
+
+	enum OverlapState {
+		OVERLAP_STATE_DIRTY = 0, // Mark processed overlaps
+		OVERLAP_STATE_INSIDE, // Mark old overlap
+		OVERLAP_STATE_ENTER, // Mark just enter overlap
+		OVERLAP_STATE_EXIT // Mark ended overlaps
+	};
+
+	struct OverlappingObjectData {
+		CollisionObjectBullet *object;
+		OverlapState state;
+
+		OverlappingObjectData()
+			: object(NULL), state(OVERLAP_STATE_ENTER) {}
+		OverlappingObjectData(CollisionObjectBullet *p_object, OverlapState p_state)
+			: object(p_object), state(p_state) {}
+		OverlappingObjectData(const OverlappingObjectData &other) {
+			operator=(other);
+		}
+		void operator=(const OverlappingObjectData &other) {
+			object = other.object;
+			state = other.state;
+		}
+	};
+
+private:
+	// These are used by function callEvent. Instead to create this each call I create if one time.
+	Variant call_event_res[5];
+	Variant *call_event_res_ptr[5];
+
+	SpaceBullet *space;
+	btGhostObject *btGhost;
```

Comment:
We use `snake_case` for identifiers throughout Godot's code base, so it would be best if you could also do the same in this module, and use e.g. `bt_ghost` (or maybe clear, `ghost_object`), `overlapping_objects`, etc.